### PR TITLE
Fix math errors in handy util

### PR DESCRIPTION
**Pull request by wakarimasent**
Fix math errors in handy utility

## Proposed changes

Stroke zone was computed incorrectly.
Also, update the tests

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by wakarimasent** [comment]
I'm not sure, how to assign this pull request to issue #77

**Comment by wakarimasent** [comment]
I'm also unable to run the tests locally. But I've updated the tests with the values which I believe are now correct.

**Comment by thefapinstructor** [comment]
Thanks @wakarimasent for the PR!  I'll be testing out the changes with the Handy Monday morning.

File: src/features/handy/utils/getStrokeSpeedAndDistance.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function getStrokeSpeedAndDistance(bps: number) {
 
   // Shorten stroke distance to increase BPS
   if (bps > fastestBps) {
-    stroke = maxStrokeSpeed / (bps * fastestBps);
+    stroke = maxStrokeLength * fastestBps / bps;
     strokeZone.min = Math.trunc(
       ((maxStrokeLength - stroke) / maxStrokeLength) * 100
```
Comment (thefapinstructor) [inline src/features/handy/utils/getStrokeSpeedAndDistance.ts:27]: I think we'll also need to adjust the zone as well.  If the bps = 4, then then stroke will be 100.  That means we are only stroking 50% of the entire strokable area.   

I think this should work, what do you think @wakarimasent?
```
    strokeZone.min = Math.trunc((maxStrokeLength - stroke) / 2);
```

Comment (wakarimasent) [reply src/features/handy/utils/getStrokeSpeedAndDistance.ts:27]: No. bps = 4 is around twice as fast as maximum bps (400/195 currently). Therefore, the stroke must be half the max stroke, which is around 100. This is correct.

Also, your proposal sets mm in the percentage field (min is from 0 to 100%), stroke and maxStrokeLength are both in mm (from 0 to 195). The current code, which divides mm by mm and scales this to 100% is correct.

Comment (wakarimasent) [reply src/features/handy/utils/getStrokeSpeedAndDistance.ts:27]: I think I might've misunderstood you question. The answer is still no. For bps =4 the stroke is 100 and current strokeZone.min will be (195-100)/195*100 ~ 50. This is correct.

Comment (thefapinstructor) [reply src/features/handy/utils/getStrokeSpeedAndDistance.ts:27]: I think you're right.  Not sure why when I tested it the faster speeds didn't work.  I'll give it another test today.

Comment (thefapinstructor) [reply src/features/handy/utils/getStrokeSpeedAndDistance.ts:27]: Test it out again and it works perfectly.  Thank you.

File: src/features/handy/utils/getStrokeSpeedAndDistance.test.ts [hunk 2]
```diff
@@ -8,48 +8,48 @@ describe("getStrokeSpeedAndDistance", () => {
...
   });
 
   it("should work for 7 bps", () => {
     const { speed, stroke, strokeZone } = getStrokeSpeedAndDistance(7);
 
     expect(speed).toBe(maxStrokeSpeed);
-    expect(stroke).toBeCloseTo(27);
-    expect(strokeZone).toEqual({ min: 85, max: 100 });
+    expect(stroke).toBeCloseTo(57);
+    expect(strokeZone).toEqual({ min: 70 max: 100 });
```
Comment (thefapinstructor) [inline src/features/handy/utils/getStrokeSpeedAndDistance.test.ts:20]: Missing a comma between min and max.

File: src/features/handy/utils/getStrokeSpeedAndDistance.test.ts [hunk 3]
```diff
@@ -8,48 +8,48 @@ describe("getStrokeSpeedAndDistance", () => {
     const { speed, stroke, strokeZone } = getStrokeSpeedAndDistance(8);
 
     expect(speed).toBe(maxStrokeSpeed);
-    expect(stroke).toBeCloseTo(24);
-    expect(strokeZone).toEqual({ min: 87, max: 100 });
+    expect(stroke).toBeCloseTo(50);
+    expect(strokeZone).toEqual({ min: 74, max: 100 });
```
Comment (thefapinstructor) [inline src/features/handy/utils/getStrokeSpeedAndDistance.test.ts:12]: Some minor adjustments to make the tests pass.

```
expect(stroke).toBeCloseTo(49);
```

File: src/features/handy/utils/getStrokeSpeedAndDistance.test.ts [hunk 4]
```diff
@@ -8,48 +8,48 @@ describe("getStrokeSpeedAndDistance", () => {
...
+    expect(strokeZone).toEqual({ min: 65, max: 100 });
   });
 
   it("should work for 5 / sec", () => {
     const { speed, stroke, strokeZone } = getStrokeSpeedAndDistance(5);
 
     expect(speed).toBe(maxStrokeSpeed);
-    expect(stroke).toBeCloseTo(39);
-    expect(strokeZone).toEqual({ min: 80, max: 100 });
+    expect(stroke).toBeCloseTo(80);
```
Comment (thefapinstructor) [inline src/features/handy/utils/getStrokeSpeedAndDistance.test.ts:35]:
```
expect(stroke).toBeCloseTo(79);
```

File: src/features/handy/utils/getStrokeSpeedAndDistance.test.ts [hunk 5]
```diff
@@ -8,48 +8,48 @@ describe("getStrokeSpeedAndDistance", () => {
...
+    expect(strokeZone).toEqual({ min: 58, max: 100 });
   });
 
   it("should work for 4 / sec", () => {
     const { speed, stroke, strokeZone } = getStrokeSpeedAndDistance(4);
 
     expect(speed).toBe(maxStrokeSpeed);
-    expect(stroke).toBeCloseTo(48);
-    expect(strokeZone).toEqual({ min: 75, max: 100 });
+    expect(stroke).toBeCloseTo(100);
```
Comment (thefapinstructor) [inline src/features/handy/utils/getStrokeSpeedAndDistance.test.ts:43]:
```
expect(stroke).toBeCloseTo(99);
```

**Review by thefapinstructor — CHANGES_REQUESTED** [review]
Just a couple tweaks to the test assertions.  Once updated I can merge.  Thank you for your contribution!

**Review by wakarimasent — COMMENTED** [review]
I've made a terrible mistake of deleting the forked repository I had made (I tried to edit the code in the browser, since I hadn't a suitable computer to check things out locally.

Now I can checkout this PR and fix the mistakes, but I can't push it anywhere.

Do you mind trying to update it?

**Comment by thefapinstructor** [comment]
> I've made a terrible mistake of deleting the forked repository I had made (I tried to edit the code in the browser, since I hadn't a suitable computer to check things out locally.
> 
> Now I can checkout this PR and fix the mistakes, but I can't push it anywhere.
> 
> Do you mind trying to update it?

Done!  Created a new https://github.com/fapinstructor/fapinstructor-client/pull/81 with these changes and the updated tests.

**Pull request closed without merging**